code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -371,3 +371,38 @@
 	desc = "An upgraded model of the basic laser gun. There seems to be some sort of slot in the handle."
 	can_take_pai = TRUE
 	origin_tech = Tc_COMBAT + "=3;" + Tc_MAGNETS + "=2;" + Tc_ENGINEERING + "=2;" + Tc_PROGRAMMING + "=4"
+
+/obj/item/weapon/gun/energy/laser/rainbow
+
+	name = "rainbow laser"
+	desc = "A fearsome gun used by clown special forces"
+	projectile_type = "/obj/item/projectile/beam/rainbow"
+	var/current_color = 1
+	var/list/color_list = list("#FF0000","#FF8C00","#FFFF00","#00FF00","#00BFFF","#0000FF","#9400D3")
+	var/fire_mode = 1
```

Comment:
This can use better naming, or usage of defines so you can do `FIRE_MODE_KILL`, `FIRE_MODE_MINDFLAY`...